### PR TITLE
Direct integration, Keycloak

### PR DESCRIPTION
**Pull request by taba90**
- Adds Keycloak UserDAO
- Adds Keycloak UserGroup 
- Adds mapping logic for Roles from keycloak to GeoStore.

### CODE REVIEW

File: src/web/app/src/main/resources/geostore-spring-security.xml [hunk 1]
```diff
@@ -169,6 +169,25 @@
...
+	<bean id="keycloakUserGroupDAO" class="it.geosolutions.geostore.services.rest.security.keycloak.KeycloakUserGroupDAO">
+		<constructor-arg ref="keycloakRESTClient"/>
+	</bean>
+	<alias name="keycloakUserGroupDAO" alias="userGroupDAO"/>
+	
+	<alias name="externalSecurityDAO" alias="securityDAO"/>
+
+	-->
+
+
```
Comment (offtherailz) [inline src/web/app/src/main/resources/geostore-spring-security.xml:190]: replace with 
```suggestion
<!--  security itegration inclusions  -->
<import resource="classpath*:security-integration-${security.integration:default}.xml"/>
```

and add files for `secuirity-integration-default.xml` and `secuirity-integration-keycloak-direct.xml`

File: src/modules/rest/api/src/main/java/it/geosolutions/geostore/services/rest/IdPLoginRest.java [hunk 2]
```diff
@@ -43,14 +45,24 @@ public interface IdPLoginRest {
 
     @GET
     @Path("/{provider}/login")
-    @Secured({"ROLE_USER", "ROLE_ADMIN", "ROLE_ANONYMOUS"})
+    @Secured({"ROLE_USER", "ROLE_ADMIN", "ROLE_GUEST", "ROLE_ANONYMOUS"})
```
Comment (offtherailz) [inline src/modules/rest/api/src/main/java/it/geosolutions/geostore/services/rest/IdPLoginRest.java:48]: We didn't support this for the moment, so please remove
```suggestion
    @Secured({"ROLE_USER", "ROLE_ADMIN", "ROLE_ANONYMOUS"})
```

**Review by offtherailz — APPROVED** [review]

**Comment by offtherailz** [comment]
going to merge next week

**Comment by ale-cristofori** [comment]
> going to merge next week

@offtherailz, keep me posted when done

**Pull request merged**